packages/sanity/src/core/search/common/deriveSearchWeightsFromType.ts

```diff
@@ -32,13 +37,27 @@ const isPtField = (type: SchemaType | undefined) =>
 const isStringField = (schemaType: SchemaType | undefined): boolean =>
   schemaType ? schemaType?.jsonType === 'string' : false
 
+const isSlugField = (schemaType: SchemaType | undefined): schemaType is SlugSchemaType => {
+  const typeChain = getTypeChain(schemaType)
+  return typeChain.some(({jsonType, name}) => jsonType === 'object' && name === 'slug')
+}
+
 const isSearchConfiguration = (options: unknown): options is SearchConfiguration =>
   isRecord(options) && 'search' in options && isRecord(options.search)
 
 function isSchemaType(input: SchemaType | CrossDatasetType | undefined): input is SchemaType {
   return typeof input !== 'undefined' && 'name' in input
 }
 
+function getFullyQualifiedPath(schemaType: SchemaType, path: string): string {
```

Comment:
Maybe we need to add a comment or maybe change the name? I'm just not sure what "fully qualified" would mean in this context 🥲 

Could it be simplified to `getFullPath` or something similar?

Reply:
Or, since it seems to just be used when the field is a `Slug` as per line 83 maybe make it related to that? 🤔

Reply:
That's a good point. ["Fully qualified name"](https://en.wikipedia.org/wiki/Fully_qualified_name) lives in my brain for some reason, so it's intuitive for me. But I understand it may not be for everybody.

In this case, I _think_ it makes sense. A selected path may point to a `slug` field, but the _full qualified path_ for searching and weighting is the field located at `slug.current`.

In a different branch, this function also automatically converts references the user has defined using dot notation e.g. `some.reference.field` becomes `some.reference->field`.

I think there is a some nuance between "full path" and "fully qualified path" but I am open to changing it. What do you think? 🙂